package/MDAnalysis/topology/base.py

```diff
@@ -39,30 +41,39 @@ class TopologyReader(IObase):
       * Be initialised with a filename
       * Return a struct dict by calling their parse function
 
-    :Raises:
-       * :exc:`IOError` upon failing to read a topology file
-       * :exc:`ValueError` upon failing to make sense of the read data
+    Raises
+    ------
+    * :exc:`IOError` upon failing to read a topology file
+    * :exc:`ValueError` upon failing to make sense of the read data
 
     .. versionadded:: 0.9.0
     .. versionchanged:: 0.9.2
        Added keyword 'universe' to pass to Atom creation.
     """
+    class __metaclass__(type):
+        def __init__(cls, name, bases, classdict):
+            type.__init__(type, name, bases, classdict)
+            try:
+                fmt = util.asiterable(classdict['format'])
+            except KeyError:
+                pass
+            else:
+                for f in fmt:
+                    _PARSERS[f] = cls
+
     def __init__(self, filename, universe=None, **kwargs):
         """Standard arguments for a TopologyReader:
```

Comment:
Hadn't harped on this yet, but for `numpy` style docs there should be no `__init__` docstring; it should instead be found in the class docstring: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#id8